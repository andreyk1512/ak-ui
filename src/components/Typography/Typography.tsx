import { TypographyProps } from './Typography.types';

export const Typography = ({ children }: TypographyProps) => <article>{children}</article>;
