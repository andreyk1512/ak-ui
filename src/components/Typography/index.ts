export * from './Typography';
export * from './Typography.types';

import { Text, Title, Paragraph } from './components';
import { Typography as OriginTypography } from './Typography';

export type TypographyProps = typeof OriginTypography & {
  Text: typeof Text;
  Title: typeof Title;
  Paragraph: typeof Paragraph;
};

export const Typography = OriginTypography as TypographyProps;
Typography.Text = Text;
Typography.Title = Title;
Typography.Paragraph = Paragraph;
