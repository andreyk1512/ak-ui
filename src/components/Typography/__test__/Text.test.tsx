import { render } from '@testing-library/react';
import { Typography } from '../Typography';

describe('Typography.Text', () => {
  it('should correctly render span element with content', () => {
    const children = 'Lorem ipsum dolor sit amet...';
    const { container } = render(<Typography.Text>{children}</Typography.Text>);

    expect(container.querySelector('span')?.textContent).toBe(children);
    expect(container.firstChild).toMatchSnapshot();
  });
});
