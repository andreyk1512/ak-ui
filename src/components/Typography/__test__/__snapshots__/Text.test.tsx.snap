// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography.Text should render the text content 1`] = `
<span
  class="css-1ti22to"
>
  This is a text component
</span>
`;
